atomix/cluster/src/test/java/io/atomix/raft/RaftResetTermAfterRestoreTest.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.1. You may not use this file
+ * except in compliance with the Zeebe Community License 1.1.
+ */
+package io.atomix.raft;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.stream.Stream;
+import org.junit.Rule;
+import org.junit.Test;
+
+// Regression test https://github.com/camunda/zeebe/issues/14509
+public class RaftResetTermAfterRestoreTest {
+  @Rule public RaftRule raftRule = RaftRule.withBootstrappedNodes(1);
+
+  @Test
+  public void shouldResetTermFromLastLogEntryIfMetastoreIsEmpty() throws Exception {
+    // given
+    raftRule.appendEntries(1);
+    final var raftServer = raftRule.getServers().stream().findFirst().get();
+    final var serverId =
+        raftServer.cluster().getLocalMember().memberId().id(); // There is only one server
+    final var termBeforeShutdown = raftServer.getTerm();
+    raftRule.shutdownServer(raftServer);
+
+    // when
+
+    // Simulate restore by deleting metastore
+    final var partitionDirectory = raftServer.getContext().getStorage().directory();
+    try (final Stream<Path> fileStream = Files.list(partitionDirectory.toPath())) {
+      final var metaFilePath =
+          fileStream
+              .filter(p -> p.getFileName().toString().endsWith("meta"))
+              .findFirst()
+              .orElseThrow(() -> new RuntimeException("No meta file found"));
+      Files.delete(metaFilePath);
+    }
+
+    raftRule.joinCluster(serverId);
+
+    // then
+    assertThat(raftRule.getLeader().orElseThrow().getTerm())
+        .describedAs("Should reset term by reading the last entry in the log")
+        // should be typically termBeforeShutdown + 1, but can be higher if there was more than one
+        // leader election.
+        .isGreaterThan(termBeforeShutdown);
```

Comment:
:question: I'm wondering if this is enough to validate this. `termBeforeShutdown` should typically be 1 (or 2?), right? Since we have a one-member cluster, there shouldn't be any leader elections. But we rely on that behavior because otherwise, the term after restart could be higher than `termBeforeShutdown` even without restoring the term from the last log entry.
So if we rely on it, we could also make it explicit and require that `termBeforeShutdown=1` and `termAfterShutdown=2`.

Reply:
That makes sense. :+1: We can pin the terms. I thought it is safer to assume it is not deterministic. But as you said, since it is a single node cluster it is fixed that the term will be 1 and 2 before and after respectively.